gooddata-sdk/gooddata_sdk/catalog/data_source/declarative_model/physical_model/pdm.py

```diff
@@ -0,0 +1,53 @@
+# (C) 2022 GoodData Corporation
+from __future__ import annotations
+
+from typing import Any
+
+from gooddata_metadata_client.model.declarative_model import DeclarativeModel
+from gooddata_metadata_client.model.declarative_tables import DeclarativeTables
+from gooddata_sdk.catalog.data_source.declarative_model.physical_model.table import CatalogDeclarativeTable
+
+
+class CatalogDeclarativePdm:
+    def __init__(self, pdm: CatalogDeclarativeTables = None):
+        self.pdm = pdm
+
+    @classmethod
+    def from_api(cls, entity: dict[str, Any]) -> CatalogDeclarativePdm:
+        pdm = CatalogDeclarativeTables.from_api(entity["pdm"])
```

Comment:
entity.get("pdm") pdm is not mandatory

maybe think about moving this logic in CatalogDeclarativeDataSource class, because there may be an object of CatalogDeclarativePdm where you will not be able to call to_api, because self.pdm will be None

Reply:
Pdm and Tables merged